tests/Scripting/Postponer.test.ts

```diff
@@ -88,6 +89,38 @@ describe('postpone - whether to show button', () => {
     });
 });
 
+describe('postpone - new task creation', () => {
+    beforeEach(() => {
+        jest.useFakeTimers();
+        jest.setSystemTime(new Date('2023-12-03'));
+    });
+
+    afterEach(() => {
+        jest.useRealTimers();
+    });
+
+    it('should postpone an overdue task to today', () => {
+        const task = new TaskBuilder().dueDate('2023-11-01').build();
+        const { postponedDate, newTasks } = createPostponedTask(task, 'dueDate', 'day', 1);
+        expect(postponedDate.format('YYYY-MM-DD')).toEqual('2023-12-03');
+        expect(newTasks.due.formatAsDate()).toEqual('2023-12-03');
+    });
+
+    it('should postpone a task scheduled today to tomorrow', () => {
+        const task = new TaskBuilder().scheduledDate('2023-12-03').build();
+        const { postponedDate, newTasks } = createPostponedTask(task, 'scheduledDate', 'day', 1);
+        expect(postponedDate.format('YYYY-MM-DD')).toEqual('2023-12-04');
+        expect(newTasks.scheduled.formatAsDate()).toEqual('2023-12-04');
+    });
+
+    it('should postpone a task that starts in the future to the next day', () => {
+        const task = new TaskBuilder().startDate('2024-03-05').build();
+        const { postponedDate, newTasks } = createPostponedTask(task, 'startDate', 'day', 1);
+        expect(postponedDate.format('YYYY-MM-DD')).toEqual('2024-03-06');
+        expect(newTasks.start.formatAsDate()).toEqual('2024-03-06');
+    });
+});
+
```

Comment:
These tests all use a pattern of `1 day`.

Might there be bugs lurking on boundary cases for units other than 1 day?

Reply:
And what happens when there is no date?

Reply:
> These tests all use a pattern of `1 day`.
> 
> Might there be bugs lurking on boundary cases for units other than 1 day?

There might indeed. However for now the postpone button postpones for 1 day only so I don't see the issue _for now_.


> And what happens when there is no date?

This should not happen since the function shall be called after `getDateFieldToPostpone()` where a check on the date presence is done.

Reply:
> This should not happen since the function shall be called after getDateFieldToPostpone() where a check on the date presence is done.

That is true now, but it may not always be true.

Plus, the current design of the code allows for this function to be called elsewhere without the caller first checking first.

Hence my question.

Reply:
> > There might indeed. However for now the postpone button postpones for 1 day only so I don't see the issue for now.
> 
> I thought that this same code was called when the user right-clicked on the postpone button, and they could defer by other increments???

Ah yes indeed you are right. The same code is called (The `QueryRenderer.getOnClickCallback()`). Excuse the confusion.

> That is true now, but it may not always be true.
> 
> Plus, the current design of the code allows for this function to be called elsewhere without the caller first checking first.

I agree that this issue exists but I don't really know how to solve it:
- jsdoc - still permissive
- encapsulate postponement in a class where the order of calls will be defined and hidden from `QueryRenderer` - definitely not in this PR
- add a check with `shouldShowPostponeButton()` - this seems redundant as it was already checked

I'm probably missing an elegant and simple solution.

Reply:
Rats. It lost my comment.

On closer inspection the tests here are mostly fine, as the finer grained tests of different date increments probably make more sense in the lower level `TasksDate.postpone()`.

I have added a failing test for when the scheduled date is inferred, though. And will push later.